dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -565,23 +575,81 @@ public static boolean versionIsGreaterThanCurrentlySupported(String semVerString
                 enhanceSemanticVersionString(LATEST_SUPPORTED_WDL_VERSION)));
     }
 
+    /**
+     * Get the version from the WDL descriptor file content. If no version is found, return the default WDL version defined by the WDL spec.
+     * If the version is invalid, return Optional.empty()
+     *
+     * @param primaryDescriptorPath The absolute path of the descriptor SourceFile to get the 'version' from
+     * @param sourceFiles           A set of SourceFiles containing the primary descriptor SourceFile and any imports
+     * @return
+     */
+    public static Optional<String> getLanguageVersion(String primaryDescriptorPath, Set<SourceFile> sourceFiles) {
+        WdlBridge wdlBridge = new WdlBridge();
+
+        Optional<SourceFile> primaryDescriptor = sourceFiles.stream().filter(sourceFile -> sourceFile.getAbsolutePath().equals(primaryDescriptorPath)).findFirst();
+        if (primaryDescriptor.isEmpty()) {
+            return Optional.empty();
+        }
+
+        final String primaryDescriptorContent = primaryDescriptor.get().getContent();
+        Map<String, String> secondaryFiles = new HashMap<>();
+        sourceFiles.stream()
+                .filter(sourceFile -> !sourceFile.getAbsolutePath().equals(primaryDescriptorPath))
+                .forEach(descriptorSourceFile -> {
+                    secondaryFiles.put(descriptorSourceFile.getAbsolutePath(), descriptorSourceFile.getContent());
+                });
+        wdlBridge.setSecondaryFiles((HashMap<String, String>)secondaryFiles);
```

Comment:
FWIW, you can define `secondaryFiles` as a `HashMap` and eliminate the cast (I think).